src/core/Bootstrap.cpp

```diff
@@ -189,6 +202,10 @@ namespace Bootstrap
             goto Cleanup;
         }
 
+        if (!AddAccessAllowedAce(pACL, ACL_REVISION, READ_CONTROL, pOwnerRightsSid)) {
```

Comment:
I'm not fully sure what this does. It reads like it just adds read permisions to the process owner. Does it also replace existing write permissions?

Reply:
It explicitly sets the permissions for the "Process Owner" to read only. Without this call, the process owner has full control by default, which allows the user to undo the other ACL's we set.